### PR TITLE
init update user, pfp

### PR DESCRIPTION
**Pull request by GraceMattern**
- update client gitignore for .env stuff
- update server typings
- update data/users
- update routes/users

### CODE REVIEW

File: server/data/users.ts [hunk 1]
```diff
@@ -40,6 +40,7 @@ export const getUserByEmail = async (email: string): Promise<UserObject> => {
  * @param {string} firebaseId The firebase id the user logged in with
  * @param {string} email The email the user logged in with
  * @param {string} name the user's display name
+ * @param {string} pfp the user's profile picture (default null)
```
Comment (roccopolimen) [inline server/data/users.ts]: pfp does not look like its passed in as a param, only put it here if it is

File: server/routes/users.ts [hunk 2]
```diff
@@ -88,11 +88,29 @@ router.post('/signUp', async (req, res) => {
...
+        if(!req.body.pfp || !checkNonEmptyString(req.body.pfp) || !checkURL(req.body.pfp)) throw new Error("[routes/users updateUser] pfp link is invalid.");
+    
+        try {
+            await updateUser(
+                req.session.user._id,
+                req.body.email,
+                req.body.name,
+                req.body.pfp
+            )
+            return res.json({ message: 'User updated.' });
```
Comment (roccopolimen) [inline server/routes/users.ts]: I would probably like to return the updated user here, just do a getUserById inside of the updateUser function probably would be best

File: server/typings.d.ts [hunk 3]
```diff
@@ -10,6 +10,7 @@ interface UserObject {
     firebaseId: string,
     email: string,
     name: string,
+    pfp: string,
```
Comment (roccopolimen) [inline server/typings.d.ts:13]: add types to both client and server to keep them consistent

**Review by roccopolimen — COMMENTED** [review]
few small comments should only take minimal effort to address

**Review by bwormser — COMMENTED** [review]
Other than the couple things Rocco said it looks good to me

**Pull request merged**